File: c++/leetcode/14. Longest Common Prefix.cpp
class Solution
{
public:
  string longestCommonPrefix(vector<string> &strs)
  {
    if (strs.size() == 1)
      return strs[0];

    string prefix = "";

    if (strs.size() < 1)
      return prefix;

    for (int i = 0; i < strs.size(); ++i)
    {
      if (strs[i].size() < strs[0].size())
      {
        auto temp = strs[0];
        strs[0] = strs[i];
        strs[i] = temp;
      }
    }

    for (int i = 0; i < strs[0].size(); ++i)
    {
      for (auto j = 0; j < strs.size(); ++j)
      {
        if (strs[0][i] != strs[j][i])
          return prefix;
      }

      prefix += strs[0][i];
    }

    return prefix;
  }
};

File: javascript/leetcode/14. Longest Common Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  if (strs.lenth == 1) return strs[0];

  let prefix = "";

  if (strs.length < 1) return prefix;

  for (let i = 0; i < strs.length; ++i) {
    if (strs[i].length < strs[0].length) {
      let temp = strs[0];
      strs[0] = strs[i];
      strs[i] = temp;
    }
  }

  for (let i = 0; i < strs[0].length; ++i) {
    for (let j = 1; j < strs.length; ++j) {
      if (strs[0][i] != strs[j][i]) return prefix;
    }

    prefix += strs[0][i];
  }

  return prefix;
};
